api/src/dataloaders/collections.rs

```diff
@@ -0,0 +1,44 @@
+use std::collections::HashMap;
+
+use async_graphql::{dataloader::Loader as DataLoader, FieldError, Result};
+use poem::async_trait;
+use sea_orm::prelude::*;
+
+use crate::{db::Connection, entities::collections, objects::Collection};
+
+#[derive(Debug, Clone)]
+pub struct ProjectLoader {
+    pub db: Connection,
+}
+
+impl ProjectLoader {
+    #[must_use]
+    pub fn new(db: Connection) -> Self {
+        Self { db }
+    }
+}
+
+#[async_trait]
+impl DataLoader<Uuid> for ProjectLoader {
+    type Error = FieldError;
+    type Value = Vec<Collection>;
+
+    async fn load(&self, keys: &[Uuid]) -> Result<HashMap<Uuid, Self::Value>, Self::Error> {
+        let collections = collections::Entity::find()
+            .filter(collections::Column::ProjectId.is_in(keys.iter().map(ToOwned::to_owned)))
+            .all(self.db.get())
+            .await?;
+
+        Ok(collections
+            .into_iter()
+            .map(|collection| (collection.project_id, collection.into()))
+            .fold(HashMap::new(), |mut acc, (project, collection)| {
```

Comment:
drop the map and do everything within the fold to avoid looping over the records twice.
```suggestion
            .fold(HashMap::new(), |mut acc, collection| {
            let project_id = collection.project_id.clone();
            let collection: Collection = collection.into();
```